source/parse.h

```diff
@@ -4197,6 +4218,10 @@ class parser
             if (!handle_logical_expression  ()) { return {}; }
             if (!handle_optional_next_clause()) { return {}; }
             if (!handle_compound_statement  ()) { return {}; }
+            if (!done() && curr().type() == lexeme::Semicolon) {
+                error("while loop body shall not end with a semicolon");
```

Comment:
"a loop body may not be followed by a semicolon (empty statements are not allowed)"